keps/1397-evictions-in-background/README.md

```diff
@@ -0,0 +1,1147 @@
+  to prefer those with eviction already in progress (or its promise)
+  to minimize workload disruptions.
+- **New prefilter algorithm to exclude in-progress evictions**: Filter out all pods
+  that have an already existing evacuation request present.
+- **Measure pending evictions**: New metric to keep count of evictions in progress/evacuation requests
+
+### Non-Goals
+
+- Implementation of any customized eviction policy: The Descheduler will only
+  acknowledge existence of pods that require different eviction handling.
+  The actual eviction (e.g. live migration) will be implemented by third parties.
+
+## Proposal
+
+This enhancement proposes the implementation of a more informed
+eviction policy for the Descheduler.
+Allowing plugins (constructing lists of pods nominated for eviction)
+to be aware of pods with eviction in progress to avoid evicting
+more than a configured limit allows.
+Reducing unnecessary evictions can improve overall workload
+distribution and save cost of bringing up new pods.
+
+### Expected Outcomes
+
+- Live migration implemented through external admission webhooks
+  are no longer perceived as eviction errors.
+  Currently, no pod with a live migration in background is evicted properly.
+  Thus resulting in eviction of all such pods from all possible nodes
+  until a limit is reached. Which is an undesirable behaviour.
+- Descheduling plugins are more aware of pods with live migration in background.
+  Thus, improving the decision making behind which pods get nominated for eviction.
+- The descheduler will no longer be responsible for direct eviction.
+  Instead, only requesting eviction and letting other components (more
+  informed and following various policies) to perform the actual eviction.
+
+### User Stories (Optional)
+
+- As a cluster admin running a K8s cluster with a descheduler,
+  I want to evict KubeVirt pods that may require live migration of VMs
+  before a pod can be  evicted. A VM live migration can take minutes
+  to complete or may require retries.
+- As an end user deploying my application on a K8s cluster with a descheduler,
+  I want to evict pods while performing pod live migration to persist
+  pod's state without storing any necessary data into pod-independent persistent storage.
+- As a developer I want to be able to implement a custom eviction policy to
+  address various company use cases that are not supported
+  by default when running a descheduler instance.
+- As a descheduler plugin developer I want to be aware of evictions
+  in progress to improve pod nomination and thus avoid unnecessary disruptions.
+- As a security professional I want to make sure any live migration
+  adhers to security policies and protects sensitive data.
+
+### Annotation vs. evacuation API based eviction
+
+The [evacuation API](https://github.com/kubernetes/enhancements/pull/4565) is expected
+to replace the [eviction API](https://kubernetes.io/docs/concepts/scheduling-eviction/api-eviction/).
```

Comment:
is there any timeline on deprecation for the eviction API?

Reply:
Not yet. It will strongly depends on when the evacuation API graduates.